src/proxy/outbound.rs

```diff
@@ -280,27 +280,6 @@ impl OutboundConnection {
             });
         }
 
-        // For case source client has enabled waypoint
-        if !source_workload.waypoint_addresses.is_empty() {
```

Comment:
Should this be a feature flag? And do any of the istio/istio tests have behavior that require client waypoints?

Reply:
@dhawton is working on the istio/istio side, I should probably mark this WIP until those changes are ready